kernel/tests/write.rs

```diff
@@ -251,3 +265,384 @@ async fn test_invalid_commit_info() -> Result<(), Box<dyn std::error::Error>> {
     ));
     Ok(())
 }
+
+fn test_read(
```

Comment:
Can we use the existing stuff in `read.rs`? I know it's not exactly the same, but I'd prefer not to re-implement read testing.

Like maybe in read.rs `read_table_data` could call a helper that took an already constructed engine, and then both this code and `read_table_data` could call that helper.

Reply:
yup had this as TODO should have commented it here - doing now!

Reply:
gonna finish in the morning

Reply:
yup unified with `read_with_execute`!